espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ViewFinder.java

```diff
@@ -155,8 +167,22 @@ public void describeTo(Description description) {
                 views = getViews(root, withText(selector), findOne);
                 break;
             case ACCESSIBILITY_ID:
-                // with content description
                 views = getViews(root, withContentDescription(selector), findOne);
+
+                // If the item is not found on the screen, use 'onData' to try
```

Comment:
why we only do it for elements located by id?

Reply:
this introduces inconsistency.

Reply:
Only elements located by content description can be targeted with a data matcher. This is like an enhancement to the accessibility id selector: If the element with the Id isn't on screen, see if we can scroll to it